src/cplus_plugin/gui/qgis_cplus_main.py

```diff
@@ -117,17 +119,25 @@ def __init__(
         self.task = None
         self.processing_cancelled = False
 
+        self.prepare_input()
+
+        # Monitors if current extents are within the pilot AOI
+        self.extent_check = PilotExtentCheck(self)
+        self.extent_check.extent_changed.connect(self.on_extent_changed)
```

Comment:
Should we monitor each time the main map canvas extent change or only when the user updates the plugin extent widget values?

Reply:
Not sure, the description in the UR is not clear on whether 'select' is map canvas change or group extent box change. But I can refactor to use the latter, please confirm. Thanks.

Reply:
Now uses signal emitted by extent widget to check extents.